flink-runtime/src/main/java/org/apache/flink/runtime/io/network/NetworkEnvironment.java

```diff
@@ -89,6 +109,8 @@ public NetworkEnvironment(
 
 		registerNetworkMetrics(metricGroup, networkBufferPool);
 
+		this.ioManager = checkNotNull(ioManager);
```

Comment:
`ioManager` looks like one more dependency for `ShuffleService` at the moment. Maybe, we could somehow get rid of it in future because it might be mot relevant for all shuffle services. I see it is used in other components. Maybe, we could create a separate `ioManager` inside netty shuffle service.

Reply:
Yes, it would be better to create all the required components inside specific `ShuffleService` implementations.

I just quickly reviewed the current usages of `IOManager`. For streaming jobs, the `BarrierBuffer` would be finally not using `IOManager` any more to cache blocked buffers in memory directly. And the `RocksDBStateBackend` only uses the `IOManager` to get path which might be done in other options I think.  So the key blocker for `IOManager` usage is from batch jobs, which relies on it heavily atm. 

Maybe we could create private `IOManager` inside `ShuffleService` as you suggested in future.